lints/lint_old_root_ca_rsa_mod_less_than_2048_bits.go

```diff
@@ -21,14 +21,12 @@ func (l *rootCaModSize) Initialize() error {
 
 func (l *rootCaModSize) CheckApplies(c *x509.Certificate) bool {
 	issueDate := c.NotBefore
-	return (c.PublicKeyAlgorithm == x509.RSA && util.IsRootCA(c) && issueDate.Before(util.RsaDate2))
+	_, ok := c.PublicKey.(*rsa.PublicKey)
+	return (c.PublicKeyAlgorithm == x509.RSA && util.IsRootCA(c) && issueDate.Before(util.RsaDate2)) && ok
```

Comment:
If we are keeping this form, then the ok should be the first check in the and statement, or even more preferably, as separate statements.